pkg/service/rtcservice.go

```diff
@@ -229,15 +229,22 @@ func (s *RTCService) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	// give it a few attempts to start session
 	var cr connectionResult
 	var initialResponse *livekit.SignalResponse
-	for i := 0; i < 3; i++ {
+	for i := 0; r.Context().Err() == nil; i++ {
 		connectionTimeout := 3 * time.Second * time.Duration(i+1)
 		ctx := utils.ContextWithAttempt(r.Context(), i)
 		cr, initialResponse, err = s.startConnection(ctx, roomName, pi, connectionTimeout)
 		if err == nil || errors.Is(err, context.Canceled) {
 			break
 		}
-		if i < 2 {
-			pLogger.Warnw("failed to start connection, retrying", err, "attempt", i)
+
+		backoffDelay := time.Duration(1<<min(i, 6)) * time.Second // exponential backoff delay. powers of 2, max 64 seconds
+		select {
+		case <-time.After(backoffDelay): // wait for backoff
```

Comment:
There is a possibility of a leak before Go 1.23. Should we use a `NewTimer` here? Go 1.23 is out I believe. So, maybe, just move to 1.23?

Reply:
I'll switch to using `time.NewTimer` just in case